spinalcordtoolbox/reports/qc2.py

```diff
@@ -0,0 +1,352 @@
+                json_data.append(json.load(file))
+
+        # Insert the QC report entries into index.html
+        path_assets = importlib.resources.files(spinalcordtoolbox.reports) / 'assets'
+        template = string.Template((path_assets / 'index.html').read_text(encoding='utf-8'))
+        # Empty the HTML file before writing, to make sure there's no leftover junk at the end
+        file_index_html.truncate()
+        file_index_html.write(template.substitute(sct_json_data=json.dumps(json_data)))
+
+        # Copy any missing assets
+        path_qc.mkdir(parents=True, exist_ok=True)
+        copy_missing(path_assets / '_assets', path_qc)
+
+    logger.info('Successfully generated the QC results in %s', str(path_result))
+    display_open(file=str(path_index_html), message="To see the results in a browser")
+
+
+def copy_missing(resource: Traversable, destination: Path):
+    """
+    Make sure that a copy of `resource` exists at `destination`,
+    by creating any missing files and directories recursively.
+    """
+    path = destination / resource.name
+    if resource.is_dir():
+        path.mkdir(exist_ok=True)
+        for sub_resource in resource.iterdir():
+            copy_missing(sub_resource, path)
+    elif resource.is_file():
+        if not path.is_file():
+            path.write_bytes(resource.read_bytes())
+    else:
+        # Some weird kind of resource? Ignore it
+        pass
+
+
+def sct_register_multimodal(
+    fname_input: str,
+    fname_output: str,
+    fname_seg: str,
+    argv: Sequence[str],
+    path_qc: str,
+    dataset: Optional[str],
+    subject: Optional[str],
+):
+    """
+    Generate a QC report for sct_register_multimodal.
+    """
+    command = 'sct_register_multimodal'
+    cmdline = [command]
+    cmdline.extend(argv)
+
+    # Axial orientation, switch between two input images
+    with qc_entry(
+        path_input=Path(fname_input).absolute(),
+        path_qc=Path(path_qc),
+        command=command,
+        cmdline=list2cmdline(cmdline),
+        plane='Axial',
+        dataset=dataset,
+        subject=subject,
+    ) as (path_background_img, path_overlay_img):
+
+        # Resample images slice by slice
+        p_resample = 0.6
+        logger.info('Resample images to %fx%f mm', p_resample, p_resample)
+        img_input = Image(fname_input).change_orientation('SAL')
+        img_input = resample_nib(
+            image=img_input,
+            new_size=[img_input.dim[4], p_resample, p_resample],
+            new_size_type='mm',
+            interpolation='spline',
+        )
+        img_output = resample_nib(
+            image=Image(fname_output).change_orientation('SAL'),
+            image_dest=img_input,
+            interpolation='spline',
+        )
+        img_seg = resample_nib(
+            image=Image(fname_seg).change_orientation('SAL'),
+            image_dest=img_input,
+            interpolation='linear',
+        )
+        img_seg.data = (img_seg.data > 0.5) * 1
```

Comment:
I think a lot of the contortions in the existing QC report code came from trying to fit these three calls to `resample_nib` into a loop. But the arguments (and post-processing) are slightly different each time, so I think it's fine to unroll the loop.